pkg/util/util.go

```diff
@@ -34,3 +37,27 @@ func IsGPUVMI(vmi *v1.VirtualMachineInstance) bool {
 	}
 	return false
 }
+
+// Check if a VMI spec requests a HostDevice
+func IsHostDevVMI(vmi *v1.VirtualMachineInstance) bool {
+	if vmi.Spec.Domain.Devices.HostDevices != nil && len(vmi.Spec.Domain.Devices.HostDevices) != 0 {
```

Comment:
`Devices` is a pointer! this can nil pointer exception...

We need to check if that's nil too.